library/core/src/main/java/com/google/android/exoplayer2/text/dvb/DvbParser.java

```diff
@@ -0,0 +1,1569 @@
+    clut.pageId = tsStream.readBits(16);
+    int remainingSegmentLength = tsStream.readBits(16);
+    clut.clutId = tsStream.readBits(8);
+    clut.clutVersionNumber = tsStream.readBits(4);
+    tsStream.skipBits(4);
+
+    remainingSegmentLength -= 2;
+    ClutEntry entry;
+    int Y, Cb, Cr, T;
+    int entryId, entryFlags;
+    while (remainingSegmentLength > 0) {
+      entryId = tsStream.readBits(8);
+      entryFlags = tsStream.readBits(8);
+
+      if ((entryFlags & 0x80) != 0) {
+        entry = clut.clutEntries2bit[entryId];
+      } else if ((entryFlags & 0x40) != 0) {
+        entry = clut.clutEntries4bit[entryId];
+      } else {
+        entry = clut.clutEntries8bit[entryId];
+      }
+
+      entry.flags = (byte) (entryFlags & 0xE1);
+      if ((entry.flags & 0x01) != 0) {
+        Y = tsStream.readBits(8);
+        Cr = tsStream.readBits(8);
+        Cb = tsStream.readBits(8);
+        T = tsStream.readBits(8);
+        remainingSegmentLength -= 6;
+      } else {
+        Y = tsStream.readBits(6) << 2;
+        Cr = tsStream.readBits(4) << 4;
+        Cb = tsStream.readBits(4) << 4;
+        T = tsStream.readBits(2) << 6;
+        remainingSegmentLength -= 4;
+      }
+
+      if (Y == 0x00) {
+        Cr = 0x00;
+        Cb = 0x00;
+        T = 0xFF;
+      }
+
+      entry.clutYCbCrT(Y, Cb, Cr, T);
+    }
+    return clut;
+  }
+
+  private ObjectData parseObjectDataSegment() {
+
+        /* Parse object data segment. ETSI EN 300 743 7.2.5
+
+                          SYNTAX                  SIZE                SEMANTICS
+        ---------------------------------------   ----   ------------------------------------------
+        object_data_segment() {
+            sync_byte                               8
+            segment_type                            8
+            page_id                                16
+            segment_length                         16
+            object_id                              16    Uniquely identifies within the page the object
+            object_version_number                   4    Indicates the version of this segment data
+            object_coding_method                    2    Specifies the method used to code the object
+            non_modifying_colour_flag               1    Indicates that the CLUT entry value '1' is a non modifying colour
+            reserved                                1
+            if (object_coding_method == '00'){
+                top_field_data_block_length        16    Specifies the number of bytes contained in his pixel-data_sub-blocks
+                bottom_field_data_block_length     16    Specifies the number of bytes contained in his pixel-data_sub-blocks
+                while(processed_length<top_field_data_block_length)
+                    pixel-data_sub-block()
+                while (processed_length<bottom_field_data_block_length)
+                    pixel-data_sub-block()
+                if (!wordaligned())
```

Comment:
This looks like it's not implemented (and needs to be). I assume you're supposed to skip 0, 1, 2 or 3 bytes so that you're a multiple of 4 from the start of object_data_segment.

Don't worry about updating the change; I'm working on some changes to merge with it and will put a fix in as part of that. But let me know if the above sounds right. It's not actually defined in the spec what the word length is, as far as I can tell!